File: venda/src/teste-oop/servico.ts
import { ItemVenda } from "./item-venda.js";

export class Servico extends ItemVenda {
  constructor(
    id: string,
    nome: string,
    preco: number,
    public duracao: number
  ) {
    super(id, nome, preco);
  }

  getTipo(): string {
    return "Serviço";
  }
}

File: venda/projeto_vendas_2/Servico.java
public class Servico extends ItemVenda {
    private static final long serialVersionUID = 1L;

    private int duracaoMinutos;

    public Servico(int id, String nome, double preco, int duracaoMinutos) {
        super(id, nome, preco);
        this.duracaoMinutos = duracaoMinutos;
    }

    public int getDuracaoMinutos() {
        return duracaoMinutos;
    }
}